examples/plugins/local/devbox.json

```diff
@@ -0,0 +1,16 @@
+{
+  "packages": [],
+  "shell": {
+    "init_hook": [
+      "echo 'Welcome to devbox!' > /dev/null"
+    ],
+    "scripts": {
+      "run_test": [
+        "./test.sh"
+      ]
+    }
+  },
+  "include": [
+    "path:my-plugin/my-plugin.json"
```

Comment:
regular plugins are specified as:
```
> {
  "includes": [
    "plugin:nginx"
  ]
}
```
https://www.jetpack.io/devbox/docs/guides/plugins/#using-plugins

what do you think of an alternative key like:
-  `plugin-path:`
- `local-plugin:`
or..even just `plugin:` and if it ends in a `.json` then we know its a local plugin.

My concern with `path` is that its a bit ambiguous in terms of "path to what?", and we could someday have `include`s that are non-plugins...

Reply:
`path` is the standard nix way to refer to local urls for inputs, which is why I used it. My thought was to reuse nix path prefixes:

```
github:
https:
path:
```

I would also be OK with dropping `path` and just allowing relative or absolute paths. So

```
my/plugin/p.json
https://myplugin.com/p.json
github:user/repo
```

@Lagoja thoughts?

Reply:
ah, that makes sense. I also like that we could later have "remote includes" specified as `https://acme.com/code/super-tool/tree/main/hammer_plugin.json`

🤔 would we ever have non-plugin includes? IIRC there was some talk of having composable `devbox.json` or other concepts that are "included".

If no, then `path` sounds great.
if yes, then `plugin-path:`, `plugin-https:`, and `plugin-github:` ?

Reply:
`includes` is intended to support devbox.json composability generally and not just plugins. But we need to merge plugin json and devbox.json.

The idea would be that including a plugin or another devbox.json is the same (so no different syntax). This would be a reason to avoid using `plugin` as part of the protocol.

Reply:
> includes is intended to support devbox.json composability generally and not just plugins. But we need to merge plugin json and devbox.json.

> The idea would be that including a plugin or another devbox.json is the same (so no different syntax). This would be a reason to avoid using plugin as part of the protocol.

I see. That makes sense as a goal. 

How do you anticipate distinguishing the current plugin-jsons versus devbox-json style plugin-jsons?

I suppose we can try to `json.Unmarshal` with current-plugin-schema and if that fails, try unmarshalling with devbox-json-style-plugin-schema.

Okay, no further concerns on this thread.

Reply:
@savil the schemas are almost identical. It's a matter of union-ing them (and possibly deprecating anything that doesn't make sense).

Reply:
I don't see why regular devbox.json should have `create_files`...

Reply:
For maintainability I would prefer to have a single schema, and could imagine good use cases: e.g. you want your devbox.json to create a process compose or config file and don't want to write a separate plugin. 

That said, it's not something this PR does, I would prefer to cross that bridge when we get to it.